### PR TITLE
Improve logging when signature checks fail

### PR DESCRIPTION
**Pull request by richvdh**
Attempts to log less useless stuff, and give clearer warnings about what's going on when there are genuine fails.

Review commit-by-commit.

### CODE REVIEW

File: synapse/federation/federation_base.py [hunk 1]
```diff
@@ -67,13 +79,10 @@ async def _check_sigs_and_hash(
...
-            logger.warning(
-                "Signature check failed for %s: %s",
-                pdu.event_id,
-                e,
-            )
-            raise
+        except InvalidEventSignatureError as e:
+            errmsg = f"event id {pdu.event_id}: {e}"
+            logger.warning("%s", errmsg)
+            raise SynapseError(403, errmsg, Codes.FORBIDDEN)
```
Comment (richvdh) [inline synapse/federation/federation_base.py]: `_check_sigs_on_pdu` now raises an `InvalidEventSignatureError` where previously it raised a `SynapseError`, so modulo slight wording tweaks to the log line and returned error, this should be identical, functionally.

File: synapse/federation/federation_client.py [hunk 2]
```diff
@@ -319,7 +323,13 @@ async def get_pdu_from_destination_raw(
             pdu = pdu_list[0]
 
             # Check signatures are correct.
-            signed_pdu = await self._check_sigs_and_hash(room_version, pdu)
+            try:
+                signed_pdu = await self._check_sigs_and_hash(room_version, pdu)
+            except InvalidEventSignatureError as e:
+                errmsg = f"event id {pdu.event_id}: {e}"
+                logger.warning("%s", errmsg)
+                raise SynapseError(403, errmsg, Codes.FORBIDDEN)
```
Comment (richvdh) [inline synapse/federation/federation_client.py:331]: this (and the three other copies of the same change) should be a non-functional change; we're just pulling the functionality out of `_check_sigs_and_hash`

File: synapse/federation/federation_client.py [hunk 3]
```diff
@@ -552,20 +562,24 @@ async def _check_sigs_and_hash_and_fetch_one(
...
-            res = await self.store.get_event(
-                pdu.event_id, allow_rejected=True, allow_none=True
+            return await self._check_sigs_and_hash(room_version, pdu)
+        except InvalidEventSignatureError as e:
+            logger.warning(
+                "Signature on retrieved event %s was invalid (%s). "
+                "Checking local store/orgin server",
+                pdu.event_id,
+                e,
             )
```
Comment (richvdh) [inline synapse/federation/federation_client.py:576]: this was the real driver for the whole PR. I wanted a clearer error message when the sig check fails in this case.

**Review by H-Shay — APPROVED** [review]
LGTM

**Pull request merged**